python/ray/tune/utils/placement_groups.py

```diff
@@ -119,24 +119,30 @@ def __init__(self,
 
         self._bind()
 
+    @property
+    def _clean_bundles(self):
+        return [{k: float(v)
+                 for k, v in bundle.items() if v != 0}
+                for bundle in self._bundles]
+
```

Comment:
Instead of making this a dynamic property, let's calculate it once in the constructor.
Also, let's keep the name `_bundles` (no need to keep the original) - and then make sure we don't access this private property anywhere else.

I.e. in the resource changing scheduler, get rid of 

```
        min_cpu_bundle = base_trial_resource._bundles[0].get("CPU", 0)
        min_gpu_bundle = base_trial_resource._bundles[0].get("GPU", 0)
```

and use `head_cpus` and a new property `head_gpus` instead

Reply:
I've moved the calculation to `__init__` and added `bundles` property, which is read only and returns a deep copy of `_bundles`.